is_sequent_provable.ml

```diff
@@ -1,10 +1,22 @@
-let is_sequent_provable_with_exceptions request_as_json =
-    let sequent = Raw_sequent.sequent_from_json request_as_json in
+let check_sequent_provability sequent =
     (Clauses.provable_sequent_as_classical sequent) &&
     (Phase.valid_sequent sequent) &&
-    (Slice.provable_sequent_as_slices sequent);;
+    (Slice.provable_sequent_as_slices sequent)
+
+let is_sequent_provable_with_exceptions request_as_json =
+    let sequent_with_notations = Sequent_with_notations.from_json request_as_json in
+    check_sequent_provability sequent_with_notations.sequent || begin
+        let cyclic_notations, sorted_acyclic_notations = Sequent_with_notations.split_cyclic_acyclic sequent_with_notations in
+        if List.length cyclic_notations > 0
+        then true
+        else if List.length sorted_acyclic_notations = 0
+            then false
+            else
+                let sequent = Sequent_with_notations.replace_all_notations_in_sequent sequent_with_notations.sequent sorted_acyclic_notations in
+                check_sequent_provability sequent
+    end;;
```

Comment:
```suggestion
    let cyclic_notations, sorted_acyclic_notations = Sequent_with_notations.split_cyclic_acyclic sequent_with_notations in
    if List.length cyclic_notations > 0
    then true
    else let sequent = Sequent_with_notations.replace_all_notations_in_sequent sequent_with_notations.sequent sorted_acyclic_notations in
         check_sequent_provability sequent
```

Reply:
You're right, it's not worth it to bring that confusion in the code.

Reply:
Done